src/beamlime/core/job.py

```diff
@@ -320,40 +397,144 @@ def reset_job(self, job_id: JobId) -> None:
         if (job := self._active_jobs.get(job_id)) is not None:
             job.reset()
         elif (job := self._scheduled_jobs.get(job_id)) is not None:
-            # Currently meaningless but might become relevant, e.g., if we add support
-            # for pausing jobs?
+            job.reset()
+        elif (job := self._stopped_jobs.get(job_id)) is not None:
             job.reset()
         else:
-            raise KeyError(f"Job {job_id} not found in active or scheduled jobs.")
+            raise KeyError(
+                f"Job {job_id} not found in active, scheduled, or stopped jobs."
+            )
+
+        # Clear error/warning state when resetting
+        self._job_error_messages.pop(job_id, None)
+        self._job_warning_messages.pop(job_id, None)
+        # Reset state to scheduled unless it's currently active
+        if job_id in self._active_jobs:
+            self._job_states[job_id] = JobState.active
+        else:
+            self._job_states[job_id] = JobState.scheduled
 
-    def push_data(self, data: WorkflowData) -> list[JobStatus]:
+    def push_data(self, data: WorkflowData) -> list[JobError]:
         """Push data into the active jobs and return status for each job."""
         self._advance_to_time(data.start_time, data.end_time)
-        job_statuses: list[JobStatus] = []
+        job_statuses: list[JobError] = []
         for job in self.active_jobs:
             status = job.add(data)
             job_statuses.append(status)
+            # Track warnings from job operations, or clear them on success
+            if status.has_error and status.error_message is not None:
+                self._job_warning_messages[job.job_id] = status.error_message
```

Comment:
I'm sort of getting confused here as to what is an error and what is a warnings (also in the `compute_results` below)
If I followed correctly, `status` here is a `JobError`. But we're putting it in the `warning_messages`...

Reply:
The name is legacy. Renaming to `JobReply` and adding comments about the warning/error distinction.